Source/events.handling/Builder/EventHandlerClassBuilder.ts

```diff
@@ -0,0 +1,109 @@
+    }
+
+    /** @inheritdoc */
+    buildAndRegister(
+        client: EventHandlersClient,
+        eventHandlers: IEventHandlers,
+        container: IContainer,
+        executionContext: ExecutionContext,
+        eventTypes: IEventTypes,
+        logger: Logger,
+        cancellation: Cancellation): void {
+
+        logger.debug(`Building event handler of type ${this._eventHandlerType.name}`);
+        const decoratedType = EventHandlerDecoratedTypes.types.find(_ => _.type === this._eventHandlerType);
+        if (decoratedType == null) {
+            logger.warning(`The event handler class ${this._eventHandlerType.name} needs to be decorated with an @${eventHandlerDecorator.name} decorator`);
+            return;
+        }
+        logger.debug(`Building ${decoratedType.partitioned ? 'partitioned' : 'unpartitioned'} event handler ${decoratedType.eventHandlerId} processing events in scope ${decoratedType.scopeId} from type ${this._eventHandlerType.name}`);
+        const methods = HandlesDecoratedMethods.methodsPerEventHandler.get(this._eventHandlerType);
+        if (methods == null) {
+            logger.warning(`There are no event handler methods to register in event handler ${this._eventHandlerType.name}. An event handler method needs to be decorated with @${handlesDecorator.name} or have the name {MethodName}`);
+            return;
+        }
+        const eventTypesToMethods = new EventTypeMap<EventHandlerSignature<any>>();
+        if (!this.tryAddAllEventHandlerMethods(eventTypesToMethods, methods, eventTypes, container, logger)) {
+            return;
+        }
+        const eventHandler = new EventHandler(decoratedType.eventHandlerId, decoratedType.scopeId, decoratedType.partitioned, eventTypesToMethods);
+        eventHandlers.register(
+            new internal.EventHandlerProcessor(
+                eventHandler,
+                client,
+                executionContext,
+                eventTypes,
+                logger), cancellation);
+    }
+
+    private tryAddAllEventHandlerMethods(eventTypesToMethods: EventTypeMap<EventHandlerSignature<any>>, methods: HandlesDecoratedMethod[], eventTypes: IEventTypes, container: IContainer, logger: Logger): boolean {
+        let allMethodsValid = true;
+        for (const method of methods) {
+            let eventType: EventType;
+            if (method.eventTypeOrId instanceof Guid || typeof method.eventTypeOrId === 'string') {
+                eventType = new EventType(
+                    EventTypeId.from(method.eventTypeOrId),
+                    method.generation != null ? Generation.from(method.generation) : undefined);
+            } else if (!eventTypes.hasFor(method.eventTypeOrId)){
+                allMethodsValid = false;
+                logger.warning(`Could not create event handler method in event handler ${this._eventHandlerType.name} because it handles an event of type ${method.eventTypeOrId.name} which is not associated to an event type`);
```

Comment:
I thought this was allowed, and that we simply deserialised the JSON content to an object and passed it as the event?